### PR TITLE
[parametric/dotnet] enable auto-instrumentation and minor code clean-up

### PR DESCRIPTION
**Pull request by lucaspimentel**
## Description

This PR makes some updates to the .NET app used for parametrics tests:
https://github.com/DataDog/system-tests/tree/main/parametric/apps/dotnet

- update nuget package references to latest versions
- enable automatic instrumentation (not used yet, but required for OTel test endpoints coming soon)
  - change `Datadog.Trace` nuget package to `Datadog.Trace.Bundle`
  - add required environment variables
- allow running app without env var `APM_TEST_CLIENT_SERVER_PORT` (easier to run as a stand-alone app)
- misc code clean-up
- added `.gitignore` and `.dockerignore` files in `parametric/apps/dotnet`

## Motivation

Automatic instrumentation in required for the .NET tracer to support the OpenTelemetry API. The rest is bonus :) 

## Additional notes

none

### CODE REVIEW

**Review by zacharycmontoya — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

File: parametric/conftest.py [hunk 1]
```diff
@@ -211,24 +211,45 @@ def dotnet_library_factory(env: Dict[str, str], container_id: str, port: str):
...
+COPY --from=build-env /client/out .
+
+# Opt-out of .NET SDK CLI telemetry (prevent unexpected http client spans)
+ENV DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+# Enable automatic instrumentation (required for OpenTelemetry tests)
+ENV CORECLR_ENABLE_PROFILING=0
+ENV CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+ENV CORECLR_PROFILER_PATH=/client/bin/Debug/net6.0/datadog/linux-x64/Datadog.Trace.ClrProfiler.Native.so
+ENV DD_DOTNET_TRACER_HOME=/client/bin/Debug/net6.0/datadog
```
Comment (zacharycmontoya) [inline parametric/conftest.py:236]:
```suggestion
ENV DD_DOTNET_TRACER_HOME=/client/datadog
```

File: parametric/conftest.py [hunk 2]
```diff
@@ -211,24 +211,45 @@ def dotnet_library_factory(env: Dict[str, str], container_id: str, port: str):
...
+WORKDIR /client
+COPY --from=build-env /client/out .
+
+# Opt-out of .NET SDK CLI telemetry (prevent unexpected http client spans)
+ENV DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+# Enable automatic instrumentation (required for OpenTelemetry tests)
+ENV CORECLR_ENABLE_PROFILING=0
+ENV CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
+ENV CORECLR_PROFILER_PATH=/client/bin/Debug/net6.0/datadog/linux-x64/Datadog.Trace.ClrProfiler.Native.so
```
Comment (zacharycmontoya) [inline parametric/conftest.py:235]:
```suggestion
ENV CORECLR_PROFILER_PATH=/client/datadog/linux-x64/Datadog.Trace.ClrProfiler.Native.so
```

Comment (lucaspimentel) [reply parametric/conftest.py:235]: oops, left over from using `dotnet run` I think? thanks!

**Comment by lucaspimentel** [comment]
Moving this to #1231 due to huge conflicts with `main`.

**Pull request closed without merging**